Civi/API/Subscriber/I18nSubscriber.php

```diff
@@ -87,12 +87,8 @@ public function onApiRespond(\Civi\API\Event\Event $event) {
    * @throws \API_Exception
    */
   public function setLocale($lcMessages, $requestId) {
-    $domain = new \CRM_Core_DAO_Domain();
-    $domain->id = \CRM_Core_Config::domainID();
-    $domain->find(TRUE);
-
     // Check if the site is multi-lingual
-    if ($domain->locales && $lcMessages) {
+    if ($lcMessages) {
```

Comment:
At the moment it looks like the only place this function is called is from up above, but it makes sense to me to have the check here? And in a unilingual install if you were to call this function you get `Warning: array_key_exists() expects parameter 2 to be array, null given in Civi\API\Subscriber\I18nSubscriber->setLocale() (line 93 of ...\Civi\API\Subscriber\I18nSubscriber.php)`.

Reply:
To be honest this code is probably going to be changed in https://github.com/civicrm/civicrm-core/pull/24174 anyway - maybe I'll close down the PR because of that - but when we tried to read the code it took far too long to figure out what it was doing